### ISSUE TITLE
prop: use Parity chain config as starting point

### ISSUE THREAD

**Issue by meowsbits** [etclabscore/eth-x-chainspec#1]
https://wiki.parity.io/Chain-specification

Pretty well documented. Pretty thorough. Parity is popular.

... Maybe some things we'd suggest to change?

**Comment by meowsbits** [comment]
@soc1c Thoughts?

**Comment by meowsbits** [comment]
Possible gripes, nitpicks, and improvements:

- `"dustProtectionTransition" - Activates Mainnet specific EIP-168 and EIP169 - Not active by default.`
  > Ideally the keys will stick to the `eip_` naming pattern.
- `validateChainIdTransition` - Optional, will be included for block 0 by default - Block before which any chain_id in the signature of a replay-protected transaction is accepted. After this transition block, the transactions’ chain_id must  vs. `eip155Transition`Optional, will be included for block 0 by default - EIP155 transition block number.
  > ... what's the difference?
- ~~where is `eip170Transition`? (`maxCodeSize` exists, yes... but no transition?)~~ `maxCodeSizeTransition`

- consensus engine param uses hardcode name as json key, should be instead a value for a standard key like `"name": "Ethash"` 
```json
	"engine": {
		"Ethash": {
			"params": {
				"minimumDifficulty": "0x020000",
```

**Comment by soc1c** [comment]
I love chain specs, but what for?

**Comment by soc1c** [comment]
a parity chain spec has everything you need to configure and run a chain, so you can use this format in any client. you don't have to reinvent the wheel. we just need to add chainspec support to other clients such as geth, multi-geth, mantis.

**Comment by soc1c** [comment]
if you change the parity chain spec format somehow, you will end up with an [xkcd 927](https://xkcd.com/927/) situation.

**Comment by meowsbits** [comment]
yea... that's what i'm figuring now, too.

Instead of reinventing (or possibly worse, "improving") the wheel, i'm just working on something from the Go side that can multi-geth can use to interact with Parity-chainspecs

That'll hopefully open some doors for x-chain,x-client tests -- which is a large part of the [reasoning for thinking about a 'formal' spec](https://github.com/etclabscore/eth-x-chain-config#why)

**Comment by meowsbits** [comment]
Although I'm not _completely_ sold that a spec needs to be a spec needs to be an "officially standardized" E/CIP-approved spec. Maybe just a tool/s reference that can pragmatically interoperate between client opinions is fine.

But 'officialness' might help in getting chain specs into [ethereum/tests](github.com/ethereum/tests), for example.

**Comment by meowsbits** [comment]
rel https://github.com/etclabscore/overflow/issues/20

**Comment by soc1c** [comment]
just port the rust impl https://github.com/paritytech/parity-ethereum/tree/master/chainspec

**Comment by meowsbits** [comment]
yep, https://github.com/etclabscore/eth-x-chainspec/blob/master/parity/config_parity.go#L10
